index.bs

```diff
@@ -267,43 +267,44 @@ for specific guidance on naming.
-so that feature detection code written for one case of unavailability
-(e.g., the feature not being implemented yet in some browsers)
-also works in other cases
-(e.g., a library being used in a non-secure context when
-the feature is limited to secure contexts).
-
-However, some features may be for
-interaction with devices that may or may not be available.
-Detection of device availability should generally be separate from feature detection.
-This helps developers who may want to handle device unavailability
-differently from lack of support for the feature.
-It also protects users; for more detail see [[#device-enumeration]].
-
-Detection should always be possible from script,
-but in some cases the feature should also be detectable
-in the language where it is used
+If appropriate, provide a way for authors to programmatically detect
+whether your feature is available,
+so that web content may gracefully handle the feature not being present.
+
+An existing feature may not be available on a page for a number of reasons, including
+- because it has not been implemented yet, or
+- because it's only available in <a href="#secure-context">secure contexts</a>, or
+- because the user denied the page permission to use the feature.
+
+Authors should need to write different code to handle each scenario,
+and should not be able to deduce which cases applied.
+
+<p class="note">
+When a feature is available but isn't feasible to use
+because a required device isn't present,
+it may be better to expose that the feature is available
+and have a separate way to detect that the device isn't.
+See [[#device-enumeration]].
+</p>
+
+Authors should always be able to detect a feature from JavaScript,
+and in some cases the feature should also be detectable
+in the language where it's used
 (such as ''@supports'' in CSS).
 
-Note that some features should not be detectable.
-In general,
-features which add capabilities that sites can use should be detectable,
-whereas features which are for end users of the web rather than for developers
-should not be detectable.
-For instance,
-sites should not be able to detect
-if [private browsing mode](https://www.w3.org/2001/tag/doc/private-browsing-modes/) is engaged,
-or if assistive technologies are being used. See [[#do-not-expose-use-of-private-browsing-mode]], [[#do-not-expose-use-of-assistive-tech]], and the [Observations on Private Browsing Modes](https://www.w3.org/2001/tag/doc/private-browsing-modes/) TAG Finding.
+In some cases, it may **not** be appropriate to allow feature detection.
+Whether the feature should be detectable or not
+should be based on the <a href="#priority-of-constituencies">user need</a>
+for the feature:
+if there is a user need or design principle which would fail
+if feature detection were available for the feature,
+then you should not support feature detection.
+
+See also:
+
+* [[#do-not-expose-use-of-private-browsing-mode]]
+* [[#do-not-expose-use-of-assistive-tech]]
```

Comment:
I would prefer to retain the general point that feature detection is for features that developers use and not for features that users use.  (In other words, it's for things that are part of the API surface of the Web.)

Reply:
OK, maybe the right distinction is just features that are exposed to developers in some way versus those that aren't.

Other examples of features that aren't exposed to developers:  find in page (although we're thinking about it), reader mode, view source, developer tools, perhaps browser UI generally (e.g., shortcut keys, ability to look at back/forward history before navigating it, etc.).